src/components/input/Input.tsx

```diff
@@ -0,0 +1,128 @@
+  unstyled: '',
+};
+
+const sizeStyles: { [K in Sizes]: string } = {
+  small: 'px-1 py-0.5 text-sm',
+  medium: 'px-2 py-1 text-base',
+  large: 'px-4 py-2 text-xl',
+};
+
+type Props = {
+  variant?: Variants;
+  inputSize?: Sizes;
+  type?: HTMLInputTypeAttribute;
+  name: string;
+  label?: string;
+  splitter?: string;
+  defaultValue?: string;
+  onChange?: ChangeEventHandler<HTMLInputElement>;
+  onComposition?: CompositionEventHandler<HTMLInputElement>;
+  onBlur?: FocusEventHandler<HTMLInputElement>;
+  placeholder?: string;
+  className?: string;
+  disabled?: boolean;
+  readOnly?: boolean;
+  autoComplete?: string;
+} & DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>;
+
+const Input = forwardRef(
+  (
+    {
+      name,
+      type = 'text',
+      label = '',
+      splitter = ':',
+      onChange,
+      onBlur,
+      onComposition,
+      placeholder = '',
+      defaultValue = '',
+      className = '',
+      disabled = false,
+      required = false,
+      variant = 'default',
+      inputSize = 'medium',
+      autoComplete = 'off',
+      readOnly = false,
+      ...props
+    }: Props,
+    ref
+  ) => {
+    const [isOnComposition, setIsOnComposition] = useState<boolean>(false);
+    const id = useId();
+
+    const handleOnChange = (event: ChangeEvent<HTMLInputElement>) => {
+      if (event.target instanceof HTMLInputElement && !isOnComposition) {
+        if (onChange) onChange(event);
+      }
+    };
+
+    const handleComposition = (event: React.CompositionEvent<HTMLInputElement>) => {
+      if (event.type === 'compositionend') {
+        if (onComposition && event.data) onComposition(event);
+        setIsOnComposition(false);
+      } else setIsOnComposition(true);
+    };
+
+    return (
+      <>
+        {label && (
+          <label
+            htmlFor={id}
+            className={clsx(
+              'mr-1 inline-flex whitespace-nowrap text-gray-50',
+              inputSize === 'large' ? 'text-xl' : '',
+              inputSize === 'small' ? 'text-sm' : ''
+            )}
+          >
+            {label} {splitter}
+          </label>
+        )}
+        <input
+          ref={() => ref && (ref as React.MutableRefObject<unknown>).current}
+          id={id}
+          name={name || id}
+          onChange={handleOnChange}
+          onCompositionStart={handleComposition}
+          onCompositionUpdate={handleComposition}
+          onCompositionEnd={handleComposition}
+          onBlur={onBlur}
+          placeholder={placeholder}
+          disabled={disabled}
+          className={clsx(
+            variant ? variantStyles?.[variant] : '',
+            inputSize ? sizeStyles?.[inputSize] : '',
+            disabled ? '!bg-opacity-50 text-gray-600' : '',
```

Comment:
should add `pointer-events-none`\
or border color will change on hover
![image](https://user-images.githubusercontent.com/18546096/211724563-9727818e-9c07-4a81-8344-d9ba321fb2e1.png)